### PR TITLE
Fixes #13900: Add documentation about rudder <-> plugin version compatibility

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/13900

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -59,20 +59,114 @@ corresponding Rudder version:
...
 
 Most of the time, there is no need to recompile a plugin for the corresponding Rudder minor version, so that
-plugins compiled on branch `branches/rudder/4.1` should be compatible with all Rudder 4.1.x versions.
+plugins compiled on branch `branches/rudder/5.1` should be compatible with all Rudder 5.1.x versions.
 
 It may happens that at some point in the Rudder maintenance cycle, a Rudder minor version introduces a
 breaking change in a plugin API or a binary incompability in a plugin ABI. In such a case, we will
 explain which plugin versions are compatible with which Rudder versions in plugin readme file.
 
+=== Summary of binary compatibility for Rudder 5.0 branch
```
Comment (amousset) [inline README.adoc]: This page is not part of Rudder doc but looks more like a developer doc, I think it should go elsewhere

Comment (fanf) [reply README.adoc]: I'm not sure I understand the remark. Should the page be in Rudder doc?

Comment (amousset) [reply README.adoc]: The general README from this rpeo in not is the doc (as it contains build instructions that not belong there, it should probable be a separate file we could directly include in the docs)

Comment (fanf) [reply README.adoc]: OH, I can do that. Does a "version-compat.adoc" file suits your need? Do you have a better name for it?

Comment (amousset) [reply README.adoc]: Sure!

File: version-compat.adoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
+=== Summary of binary compatibility for Rudder 5.0 branch
```
Comment (amousset) [inline version-compat.adoc]: Could you make it a level 1 title instead, it will be handled by inclusion code

**Comment by fanf** [comment]
Commit modified

**Comment by amousset** [comment]
OK, squash merging this PR

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**